pkg/app/config.go

```diff
@@ -74,6 +84,40 @@ func CreateConfig(configFile string) (Config, error) {
 		config.Openshift.Namespace = string(dat)
 	}
 
+	for regCount, reg := range config.Registry {
+		if reg.AuthType == "secret" {
+			data, err = getSecretData(reg.AuthName, config.Openshift.Namespace)
+			if err != nil {
+				fmt.Println("Unable to get secret data")
+				// NEW ERROR
+				return Config{}, err
+			}
+			var username = strings.TrimSpace(string(data["username"]))
+			var password = strings.TrimSpace(string(data["password"]))
+
+			if username == "" || password == "" {
+				fmt.Printf("Unable to find credentials in secret: %s\n", reg.AuthName)
+				return Config{}, err
+			}
+
+			config.Registry[regCount].User = username
+			config.Registry[regCount].Pass = password
+
+		} else if reg.AuthType == "file" {
+			if dat, err = ioutil.ReadFile(reg.AuthName); err != nil {
+				return Config{}, err
+			}
+			if err = yaml.Unmarshal(dat, &regCreds); err != nil {
+				return Config{}, err
+			}
+			config.Registry[regCount].User = regCreds.Username
+			config.Registry[regCount].Pass = regCreds.Password
+
+		} else {
+			// ERROR
```

Comment:
Should we still allow the user to specify the user/pass in the config and use them here? This way we don't break old brokers and the real motivation was this for when we are running in openshift.

Reply:
@shawn-hurley that was how I previously implemented this before the refactoring :). That worked fine but required entrypoint changes. Jesus' brought up some good feedback that it would be cleaner to read the secret from the broker itself. We also control the template so its arbitrary which one we pick.

Reply:
@dymurray If you are saying allow the user to use the old configuration values if they want, that's ok with me. If someone chooses to put a password in a config it is up to them.